### PR TITLE
Add javadoc to the Graph API

### PR DESCRIPTION
**Pull request by mathbagu**
**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [ ] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

### CODE REVIEW

File: math/src/main/java/com/powsybl/math/graph/UndirectedGraphImpl.java [hunk 1]
```diff
@@ -115,6 +115,9 @@ private void checkEdge(int e) {
         }
     }
 
+    /**
+     * {@inheritDoc}
+     */
```
Comment (sylvlecl) [inline math/src/main/java/com/powsybl/math/graph/UndirectedGraphImpl.java]: True for all other methods : inheriting doc is the default behaviour, so this is useful only when you want to add some javadoc.

File: math/src/main/java/com/powsybl/math/graph/UndirectedGraphImpl.java [hunk 2]
```diff
@@ -390,6 +472,18 @@ public void traverse(int v, Traverser traverser) {
         return paths;
     }
 
+    /**
+     * This method is called by {@link #findAllPaths(int, Function, Function, TIntArrayList, BitSet, List)} each time a vertex is traversed.
+     *
```
Comment (sylvlecl) [inline math/src/main/java/com/powsybl/math/graph/UndirectedGraphImpl.java]: The javadoc suggests that the name is not very well chosen :) Not a big issue though...

File: math/src/main/java/com/powsybl/math/graph/UndirectedGraph.java [hunk 3]
```diff
@@ -14,62 +14,229 @@
...
+     */
     void traverse(int v, Traverser traverser);
 
+    /**
+     * Find all paths from the specified vertex.
+     * This method relies on two functions to stop the traverse when the target vertex is found or when an edge must not be traversed.
+     *
+     * @param from the vertex index where the traverse has to start.
+     * @param pathComplete a function that returns true when the target vertex is found.
+     * @param pathCanceled a function that returns true when the edge must not be traversed.
```
Comment (sylvlecl) [inline math/src/main/java/com/powsybl/math/graph/UndirectedGraph.java]: Not part of the PR, but there is a typo in "cancelled".

File: math/src/main/java/com/powsybl/math/graph/UndirectedGraph.java [hunk 4]
```diff
@@ -14,62 +14,229 @@
...
+     *
+     * @return a {@link List} containing the attached values to the edges between the vertices v1 and v2.
+     */
     List<E> getEdgeObjects(int v1, int v2);
 
+    /**
+     * Traverse the entire graph, starting at the specified vertex v.
+     * This method relies on a {@link Traverser} instance to know if the traverse of the graph should continue or stop.
+     * This method throws a {@link com.powsybl.commons.PowsyblException} if the encountered table size is lesser than the vertex count.
+     *
```
Comment (sylvlecl) [inline math/src/main/java/com/powsybl/math/graph/UndirectedGraph.java]: "is less than the maximum vertex index" instead of vertex count

File: math/src/main/java/com/powsybl/math/graph/UndirectedGraph.java [hunk 5]
```diff
@@ -14,62 +14,229 @@
...
 
+    /**
+     * Return an {@link Iterable} to iterate over the attached values to the edges.
+     *
+     * @return an {@link Iterable} to iterate over the attached values to the edges.
+     */
     Iterable<E> getEdgesObject();
 
+    /**
+     * Return a {@link Stream} to iterate over the attached values to the edges.
```
Comment (sylvlecl) [inline math/src/main/java/com/powsybl/math/graph/UndirectedGraph.java]: I think "the values attached to the edges" sounds better.

File: math/src/main/java/com/powsybl/math/graph/UndirectedGraph.java [hunk 6]
```diff
@@ -14,62 +14,229 @@
...
     int getEdgeVertex2(int e);
 
+    /**
+     * Remove all the vertices of this graph.
+     * This method throws a {@link com.powsybl.commons.PowsyblException} if edges exist.
+     */
     void removeAllVertices();
 
+    /**
+     * Return an {@link Iterable} to iterate over the attached values to the edges.
```
Comment (sylvlecl) [inline math/src/main/java/com/powsybl/math/graph/UndirectedGraph.java]: I think "the values attached to the edges" sounds better.

File: math/src/main/java/com/powsybl/math/graph/UndirectedGraph.java [hunk 7]
```diff
@@ -14,62 +14,229 @@
...
 
+    /**
+     * Return a {@link Stream} to iterate over the attached values to the vertices.
+     *
+     * @return a {@link Stream} to iterate over the attached values to the vertices.
+     */
     Stream<V> getVertexObjectStream();
 
+    /**
+     * Return the attached value to the specified vertex.
```
Comment (sylvlecl) [inline math/src/main/java/com/powsybl/math/graph/UndirectedGraph.java]: I think "the value attached to the specified vertex" sounds better.

File: math/src/main/java/com/powsybl/math/graph/UndirectedGraph.java [hunk 8]
```diff
@@ -14,62 +14,229 @@
...
 
+    /**
+     * Return an {@link Iterable} to iterate over the attached values to the vertices.
+     *
+     * @return an {@link Iterable} to iterate over the attached values to the vertices.
+     */
     Iterable<V> getVerticesObj();
 
+    /**
+     * Return a {@link Stream} to iterate over the attached values to the vertices.
```
Comment (sylvlecl) [inline math/src/main/java/com/powsybl/math/graph/UndirectedGraph.java]: I think "the values attached to the vertices" sounds better.

File: math/src/main/java/com/powsybl/math/graph/UndirectedGraph.java [hunk 9]
```diff
@@ -14,62 +14,229 @@
...
+    /**
+     * Return the maximum count of vertices, including available vertices.
+     * The effective count of vertices is given by {@link #getVertexCount()}.
+     *
+     * @return the maximum count of vertices.
+     */
     int getMaxVertex();
 
+    /**
+     * Return an {@link Iterable} to iterate over the attached values to the vertices.
```
Comment (sylvlecl) [inline math/src/main/java/com/powsybl/math/graph/UndirectedGraph.java]: I think "the values attached to the vertices" sounds better.

File: math/src/main/java/com/powsybl/math/graph/UndirectedGraph.java [hunk 10]
```diff
@@ -14,62 +14,229 @@
...
     int[] getEdges();
 
+    /**
+     * Return the indexes of the vertices.
+     * @return the indexes of the vertices.
+     */
     int[] getVertices();
 
+    /**
+     * Return the maximum count of vertices, including available vertices.
```
Comment (sylvlecl) [inline math/src/main/java/com/powsybl/math/graph/UndirectedGraph.java]: Maybe we should say the maximum vertex **index**, because "count" sounds as if there is no "hole".

File: math/src/main/java/com/powsybl/math/graph/UndirectedGraph.java [hunk 11]
```diff
@@ -14,62 +14,229 @@
...
     void removeAllEdges();
 
+    /**
+     * Return the number of edges.
+     * @return the number of edges.
+     */
     int getEdgeCount();
 
+    /**
+     * Return the indexes of the edges.
```
Comment (sylvlecl) [inline math/src/main/java/com/powsybl/math/graph/UndirectedGraph.java]: I would say "indices" for the plural form, but it seems indexes is used too. However in technical / math contexts it would mainly be "indices".
https://writingexplained.org/indexes-or-indices-difference

File: math/src/main/java/com/powsybl/math/graph/UndirectedGraph.java (hunk 3 above)
Comment (mathbagu) [reply math/src/main/java/com/powsybl/math/graph/UndirectedGraph.java]: Fixed

File: math/src/main/java/com/powsybl/math/graph/UndirectedGraphImpl.java (hunk 2 above)
Comment (mathbagu) [reply math/src/main/java/com/powsybl/math/graph/UndirectedGraphImpl.java]: Which name? The method name?

Comment (sylvlecl) [reply math/src/main/java/com/powsybl/math/graph/UndirectedGraphImpl.java]: Yes that's what I meant, because from the javadoc it does not sound like it finds all paths from somewhere, but after looking into what it does it's not so bad.
Maybe the javadoc could say "Recursively finds all paths starting from specified vertex of specified edge" or something like that, together with "is used by findAllPaths".

Not really important anyway to be honest :)

File: math/src/main/java/com/powsybl/math/graph/UndirectedGraph.java [hunk 12]
```diff
@@ -14,62 +14,255 @@
...
+     * This method throws a {@link com.powsybl.commons.PowsyblException} if the vertex doesn't exist or if an edge is connected to this vertex.
+     *
+     * @param v the vertex index to remove.
+     * @return the value attached to the vertex.
+     */
     V removeVertex(int v);
 
+    /**
+     * Return the number of vertices.
+     * As the contiguity of vertices is not mandatory, the number of vertices can be less than the higher vertex index.
```
Comment (MioRtia) [inline math/src/main/java/com/powsybl/math/graph/UndirectedGraph.java]: typo
> less than the high**est** vertex index

File: math/src/main/java/com/powsybl/math/graph/UndirectedGraph.java [hunk 13]
```diff
@@ -14,62 +14,255 @@
...
+     *
+     * @return the number of vertices.
+     */
     int getVertexCount();
 
+    /**
+     * Create an edge between the two specified vertex and notify the {@link UndirectedGraphListener}s.
```
Comment (MioRtia) [inline math/src/main/java/com/powsybl/math/graph/UndirectedGraph.java]: two specified **vertices** ?

File: math/src/main/java/com/powsybl/math/graph/UndirectedGraph.java [hunk 14]
```diff
@@ -14,62 +14,255 @@
...
+    /**
+     * Remove a {@link UndirectedGraphListener} to stop listening the graph changes.
+     *
+     * @param l the listener to remove.
+     */
     void removeListener(UndirectedGraphListener l);
 
+    /**
+     * Prints the entire graph to the specified {@link PrintStream}.
+     * The printing relies on the two specified function to render the values attached to a vertex or an edge.
```
Comment (MioRtia) [inline math/src/main/java/com/powsybl/math/graph/UndirectedGraph.java]: typo:
> function**s**
